test/e2e/suite/command/custom_header.go

```diff
@@ -16,10 +16,78 @@ limitations under the License.
 package command
 
 import (
+	"fmt"
+
 	. "github.com/onsi/ginkgo/v2"
+	"oras.land/oras/test/e2e/internal/testdata/foobar"
+	"oras.land/oras/test/e2e/internal/testdata/multi_arch"
 	. "oras.land/oras/test/e2e/internal/utils"
 )
 
+var _ = Describe("Common registry users:", func() {
+	var (
+		FoobarHeaderInput = "Foo:bar"
+		FoobarHeader      = "\"Foo\": \"bar\"\n"
+		AbHeaderInput     = "A: b"
+		AbHeader          = "\"A\": \" b\"\n"
+	)
+	When("custom header is provided", func() {
+		It("attach", func() {
+			testRepo := attachTestRepo("simple")
```

Comment:
E2E tests are all running in parallel, to make sure different specs are segregated from each other, use different repository for testing.

In this PR, add a new `headerTestRepo` func like [others](https://github.com/oras-project/oras/blob/main/test/e2e/suite/command/attach.go#L35) and use it for `attach`, `push` and `tag` tests.

Reply:
Added a new test repo for headers.